src/selectors/entities/channel_categories.ts

```diff
@@ -0,0 +1,375 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {createSelector} from 'reselect';
+import shallowEquals from 'shallow-equals';
+
+import {General, Preferences} from '../../constants';
+import {CategoryTypes} from '../../constants/channel_categories';
+
+import {getCurrentChannelId, getMyChannelMemberships} from 'selectors/entities/channels';
+import {getCurrentUserLocale} from 'selectors/entities/i18n';
+import {getLastPostPerChannel} from 'selectors/entities/posts';
+import {getMyPreferences, getTeammateNameDisplaySetting, shouldAutocloseDMs} from 'selectors/entities/preferences';
+import {getCurrentUserId} from 'selectors/entities/users';
+
+import {Channel, ChannelMembership} from 'types/channels';
+import {ChannelCategory} from 'types/channel_categories';
+import {GlobalState} from 'types/store';
+import {UserProfile} from 'types/users';
+import {IDMappedObjects, RelationOneToOne} from 'types/utilities';
+
+import {getUserIdFromChannelName, isFavoriteChannel, isUnreadChannel} from 'utils/channel_utils';
+import {getPreferenceKey} from 'utils/preference_utils';
+import {displayUsername} from 'utils/user_utils';
+
+export function getCategoryIdsForTeam(state: GlobalState, teamId: string): string[] | undefined {
+    return state.entities.channelCategories.orderByTeam[teamId];
+}
+
+export function makeGetCategoriesForTeam(): (state: GlobalState, teamId: string) => ChannelCategory[] {
+    return createSelector(
+        getCategoryIdsForTeam,
+        (state: GlobalState) => state.entities.channelCategories.byId,
+        (categoryIds, categoriesById) => {
+            if (!categoryIds) {
+                return [];
```

Comment:
Out of curiosity,  when would we not have  `getCategoryIdsForTeam`?

Reply:
It might happen in the future when these are loaded from the server, but they should always be set now. It's also just error proofing since you could call the selector with any random string